pkg/checker/checker.go

```diff
@@ -59,15 +68,23 @@ func (c *VersionChecker) Check(kclPkg pkg.KclPkg) error {
 }
 
 // SumChecker validates the dependencies checksum in kclPkg.
-type SumChecker struct{}
+type SumChecker struct {
+	kpmcli *client.KpmClient
```

Comment:
Hi @NishantBansal2003 😄 , you can also try to transfer the process of getting Sum from KpmClient to SumChecker, this structure is not suitable at present, I am also refactoring this part of the structure, the next best state is in KpmClient, including SumChecker, Instead of including KpmClient in SumChecker, SumChecker provides verification capabilities.

Reply:
Since the `kpmClient` is used to get the settings using `kpmcli.GetSettings()`, through which we are getting `DefaultOciRegistry, DefaultOciRepo, and Credential`, and since `kpmClient's` settings are initialized from `settings.GetSettings()`, can I use `settings.GetSettings()` directly in `SumChecker` instead of using it from `kpmcli.GetSettings()`? This way, the dependency of getting `trustedSum` from `kpmClient` will be moved to `SumChecker`.

Reply:
> Since the `kpmClient` is used to get the settings using `kpmcli.GetSettings()`, through which we are getting `DefaultOciRegistry, DefaultOciRepo, and Credential`, and since `kpmClient's` settings are initialized from `settings.GetSettings()`, can I use `settings.GetSettings()` directly in `SumChecker` instead of using it from `kpmcli.GetSettings()`? This way, the dependency of getting `trustedSum` from `kpmClient` will be moved to `SumChecker`.

Hi @NishantBansal2003 😄  You can set `Settings` as a member of the `SumChecker` structure when initializing the `SumChecker` instance.